### PR TITLE
Konductor responds with RFC 7807 for fatal errors

### PR DESCRIPTION
**Pull request by raresvla**
## Description

- Fixed tests to properly assert the new error response
- Opt-out is no longer considered an error (is a warning)

## Motivation and Context

Updates test asserts to comply with upcoming Konductor changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by Aaronius** [comment]
Hey @raresvla, just so you're aware, you should have admin rights to this repo so you don't have to work on your personal fork. In fact, we encourage you working on the main fork because it's easier for us to check out and add commits to your branch.

File: test/functional/specs/Privacy/IAB/C224677.js [hunk 1]
```diff
@@ -67,11 +65,17 @@ test("Test C224677: Call setConsent when purpose 10 is FALSE", async () => {
   const identityHandle = response.getPayloadsByType("identity:result");
   await t.expect(identityHandle.length).eql(2);
 
-  // Event calls going forward should be Opted-Out because AAM opts out consents with no purpose 10.
-  const errorMessage = await getErrorMessageFromSendEvent();
+  // 3. Event calls going forward should be Opted-Out because AAM opts out consents with no purpose 10.
```
Comment (Aaronius) [inline test/functional/specs/Privacy/IAB/C224677.js:68]: Grammatically, "Opted-Out" here should be "opted out" (it's the past tense of a transitive verb).

File: test/functional/specs/Privacy/IAB/C224677.js [hunk 2]
```diff
@@ -67,11 +65,17 @@ test("Test C224677: Call setConsent when purpose 10 is FALSE", async () => {
...
+  await triggerEvent();
+  const rawEventResponse = JSON.parse(
+    getResponseBody(networkLogger.edgeEndpointLogs.requests[0])
+  );
+  const eventResponse = createResponse(rawEventResponse);
 
-  await t.expect(errorMessage).contains("EXEG-0301-403");
+  // 4. And a warning message should be returned, confirming the Opted-Out
```
Comment (Aaronius) [inline test/functional/specs/Privacy/IAB/C224677.js:75]: Grammatically, "Opted-Out" here should be "opt-out" (it's a noun).

File: test/functional/specs/Privacy/IAB/C224678.js [hunk 3]
```diff
@@ -97,7 +96,11 @@ test.skip("Test C224678: Passing a negative Consent in the sendEvent command", a
     .expect(errorMessage)
     .notOk("Event returned an error when we expected it not to.");
 
-  // 5. Events should be blocked going forward because we are opted out.
+  // 5. But returns a warning message confirming the Opted-Out
```
Comment (Aaronius) [inline test/functional/specs/Privacy/IAB/C224678.js:99]: "Opted-Out" should be "opt-out"

**Review by Aaronius — APPROVED** [review]
I believe this all makes sense to me and looks good. Thanks for making the updates @raresvla!

**Comment by raresvla** [comment]
@Aaronius I've addressed your comments, thank you! 

@jfkhoury / @shammowla -- I see build that is failing here, although is passing locally. Is this a know issue?

**Review by jfkhoury — APPROVED** [review]
Thank you @raresvla !

@shammowla Can you please look into the tests?

**Comment by shammowla** [comment]
Hey @raresvla, I was able to pinpoint issue with the tests failing on GH actions. I cloned your commits and added it to this PR: https://github.com/adobe/alloy/pull/642

The required keys and environment variables didn't pass to GH actions. Also, the unit tests karma runner was creating a separate tunnel to browserstack which caused the unit tests to fail. 

Both the unit tests and functional test passed with the mentioned changes. 
https://github.com/adobe/alloy/actions/runs/409042942

**Comment by Aaronius** [comment]
@raresvla, thanks for your help on this PR. I'm going to close it since it's been superseded by https://github.com/adobe/alloy/pull/642 (your changes are reflected there).

**Pull request closed without merging**